services/venice-controller/src/main/java/com/linkedin/venice/controller/server/StoresRoutes.java

```diff
@@ -506,6 +506,12 @@ public Route deleteStore(Admin admin) {
     return new VeniceRouteHandler<TrackableControllerResponse>(TrackableControllerResponse.class) {
       @Override
       public void internalHandle(Request request, TrackableControllerResponse veniceResponse) {
+        // This is to limit the manual store deletion from admin tool without https to a specific allow list users.
+        if (!isSslEnabled()
+            && !checkIsAllowListUser(request, veniceResponse, () -> isAllowListUserForStoreDeletion(request))) {
+          return;
```

Comment:
We should fail the request and send bad request error code to user.

Reply:
The `checkedIsAllowListUser` will mark the request as Bad Request: 
```
  protected boolean checkIsAllowListUser(
      Request request,
      ControllerResponse veniceResponse,
      BooleanSupplier isAllowListUser) {
    if (!isAllowListUser.getAsBoolean()) {
      veniceResponse.setError(ACL_CHECK_FAILURE_WARN_MESSAGE_PREFIX + request.url());
      veniceResponse.setErrorType(ErrorType.BAD_REQUEST);
      return false;
    }
    return true;
  }
```